### PR TITLE
TowerOfHanoi-recursion

### PR DESCRIPTION
**Pull request by sreshtech**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/backtracking/TowerOfHanoi.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        
+    }
+
+    public static void main(String[] args) {
+        int n=10;//number of disk, you can change it or make it user input using Scanner
+        
+        towerOfHanoi(n,"S", "H", "D");
+
+        
+    }
```
Comment (vil02) [inline src/main/java/com/thealgorithms/backtracking/TowerOfHanoi.java:31]: Please remove the `main` method and instead add some proper tests. Good examples to look at:
- [`Factorial`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/main/java/com/thealgorithms/maths/Factorial.java) and [`FactorialTest`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/test/java/com/thealgorithms/maths/FactorialTest.java),
- [`FindMax`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/main/java/com/thealgorithms/maths/FindMax.java) and [`FindMaxTest`](https://github.com/TheAlgorithms/Java/blob/566c27a996f48a94916170c159ae735546b39c42/src/test/java/com/thealgorithms/maths/FindMaxTest.java).


```suggestion
```

File: src/main/java/com/thealgorithms/backtracking/TowerOfHanoi.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.backtracking;
+/*The problem consists of three rods and a number of disks of different sizes, which can be slid onto any rod. The puzzle starts with the disks in a neat stack in ascending order of size on one rod, the source rod, with the largest disk at the bottom and the smallest at the top. The goal is to move the entire stack to another rod, the target rod, subject to the following rules:
+
+Only one disk can be moved at a time.
+A disk can only be placed on top of a larger disk or an empty rod. */
+
+public class TowerOfHanoi {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/backtracking/TowerOfHanoi.java:7]: Let's make it a _proper utility class_.

```suggestion
public final class TowerOfHanoi {
    private TowerOfHanoi() {
    }
```

**Review by vil02 — CHANGES_REQUESTED** [review]
Always please make sure that the CI passes.

**Comment by sreshtech** [comment]
> Always please make sure that the CI passes and _check_ all of the _boxes_ in the PR (i.e. change `[ ]` to `[x]`).

DONE
Removed the main method and  added some proper tests.
PLEASE  ask to resolve if more issues. thank you!

**Comment by BamaCharanChhandogi** [comment]
@vil02 The Tower of Hanoi is already available.

**Comment by BamaCharanChhandogi** [comment]
I am closing this issue.

**Pull request closed without merging**